Tests/SwiftParserTest/ParserTests.swift

```diff
@@ -1,8 +1,52 @@
+import Dispatch
 import XCTest
 import SwiftSyntax
 import SwiftParser
 
 public class ParserTests: XCTestCase {
+  /// Run a single parse test.
+  func runParseTest(fileURL: URL, checkDiagnostics: Bool) throws {
+    let fileContents = try String(contentsOf: fileURL)
+    let parsed = try Parser.parse(source: fileContents)
+    AssertStringsEqualWithDiff("\(parsed)", fileContents)
+
+    if !checkDiagnostics {
+      return
+    }
+
+    let diagnostics = ParseDiagnosticsGenerator.diagnostics(for: parsed)
+    if !diagnostics.isEmpty {
+      var locationAndDiagnostics: [String] = []
+      let locationConverter = SourceLocationConverter(file: fileURL.lastPathComponent, tree: parsed)
+      for diag in diagnostics {
+        let location = diag.location(converter: locationConverter)
+        let message = diag.message
+        locationAndDiagnostics.append("\(location): \(message)")
+      }
+      XCTFail("""
```

Comment:
Could you pass `file` and `line` from `runParserTests` to this `XCTFail` and the `AssertStringsEqualWithDiff` above?